### PR TITLE
Reorganized Generic HTTP Config Connector README for easier readability

### PR DESCRIPTION
**Pull request by mdodell**
### What does this PR do?
- Reorganized Generic HTTP Config Connector README for easier readability

### What ticket does this PR close?
Connected to #1286 

#### Change log
- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [x] The changes in this PR do not require tests

#### Documentation
- [ ] This PR does not require updating any documentation, or
- [x] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs

#### (For releases only) Manual tests
- [ ] Manually tested [Keychain provider](https://github.com/cyberark/secretless-broker/tree/master/test/providers/keychain)
- [ ] Manually run [Kubernetes-Conjur demo](https://github.com/conjurdemos/kubernetes-conjur-demo) with a local Secretless Broker image build of your branch

### CODE REVIEW

File: examples/generic_connector_configs/README.md [hunk 1]
```diff
@@ -1,30 +1,232 @@
...
+    </li>
+    <li>
+      Use the provided cacert.pem file on the Splunk docker container
+      for my certificate, and write it to the local machine
+    </li>
+    <code>docker exec -it splunk sudo cat /opt/splunk/etc/auth/cacert.pem > myLocalSplunkCertificate.pem</code>
+    <li>
+      Set a variable in the terminal named
+      <code>
+        <a href="https://github.com/cyberark/secretless-broker/pull1180">SECRETLESS_HTTP_CA_BUNDLE</a>
```
Comment (JakeQuilty) [inline examples/generic_connector_configs/README.md:211]: 404

File: examples/generic_connector_configs/README.md [hunk 2]
```diff
@@ -1,30 +1,232 @@
 # Generic HTTP Connector Example Configurations
 
+## Table of Contents
```
Comment (JakeQuilty) [inline examples/generic_connector_configs/README.md:3]: I think this should follow the format of the [README ToC](https://github.com/cyberark/secretless-broker/blob/master/README.md#table-of-contents). I don't think it needs the links to itself and the main title. I also think it shouldn't link to the examples and "how tos" in all of them. 

```suggestion
## Table of Contents
- [Sample Configurations](#sample-configurations)
   - [Elasticsearch API](#elasticsearch-api)
   - [Github API](#github-api)
   - [OAuth 2.0 API](#oauth-20-api)
   - [Slack Web API](#slack-web-api)
   - [Splunk API](#splunk-api)
- [Contributing](#contributing)
```

File: examples/generic_connector_configs/README.md [hunk 3]
```diff
@@ -1,30 +1,232 @@
...
+    + [How to use this connector](#how-to-use-this-connector-2)
+  * [Slack Web API](#slack-web-api)
+    + [How to use this connector](#how-to-use-this-connector-3)
+    + [Example Usage](#example-usage-2)
+  * [Splunk API](#splunk-api)
+    + [How to use this connector](#how-to-use-this-connector-4)
+    + [Example Usage](#example-usage-3)
+- [Contributing](#contributing)
+
+# Sample Configurations
```
Comment (JakeQuilty) [inline examples/generic_connector_configs/README.md:23]:
```suggestion
## Sample Configurations
```

File: examples/generic_connector_configs/README.md [hunk 4]
```diff
@@ -1,30 +1,232 @@
...
-
-## Sample Configurations
+[documented instructions](https://docs.secretless.io/Latest/en/Content/References/connectors/scl_handlers-https.htm#Manageservercertificates) for adding the
+target’s CA to Secretless’ trusted certificate pool.
 
 > Note: The following examples use the [Keychain provider](https://docs.cyberark.com/Product-Doc/OnlineHelp/AAM-DAP/11.3/en/Content/References/providers/scl_keychain.htm?TocPath=Fundamentals%7CSecretless%20Pattern%7CSecret%20Providers%7C_____5).
 > Replace the service prefix `service#` with an appropriate service
 > or use a different provider as needed.
+___
+## Elasticsearch API
```
Comment (JakeQuilty) [inline examples/generic_connector_configs/README.md:39]:
```suggestion
### Elasticsearch API
```

File: examples/generic_connector_configs/README.md [hunk 5]
```diff
@@ -1,30 +1,232 @@
...
+This example can be used to interact with [Elasticsearch's API](https://www.elastic.co/guide/en/elasticsearch/reference/current).
+
+The configuration file for the Elasticsearch API can be found at
+[elasticsearch_secretless.yml](./elasticsearch_secretless.yml).
+
+### How to use this connector
```
Comment (JakeQuilty) [inline examples/generic_connector_configs/README.md:45]:
```suggestion
#### How to use this connector
```

File: examples/generic_connector_configs/README.md [hunk 6]
```diff
@@ -1,30 +1,232 @@
...
+* Edit the supplied configuration to get your Elasticsearch
+[API Key](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-create-api-key.html) or
+[OAuth token](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-get-token.html)
+* Run Secretless with the supplied configuration(s)
+* Query the Elasticsearch API using `http_proxy=localhost:9020 curl <Elasticsearch Endpoint URL>/{Request}`
+
+### Example Usage
```
Comment (JakeQuilty) [inline examples/generic_connector_configs/README.md:52]:
```suggestion
#### Example Usage
```

File: examples/generic_connector_configs/README.md [hunk 7]
```diff
@@ -1,30 +1,232 @@
...
+      <br />
+      -f examples/generic_connector_configs/elasticsearch.yml
+    </code>
+    <li>Query the ElasticSearch API using <code>http_proxy=localhost:9020 curl <Elasticsearch Endpoint URL>/{Request}</code></li>
+  </ol>
+</details>
+
+___
+
+## Github API
```
Comment (JakeQuilty) [inline examples/generic_connector_configs/README.md:73]:
```suggestion
### Github API
```

File: examples/generic_connector_configs/README.md [hunk 8]
```diff
@@ -1,30 +1,232 @@
...
+This example can be used to interact with [Github's API](https://developer.github.com/v3/).
+
+The configuration file for the Github API can be found at [github_secretless.yml](./github_secretless.yml).
+
+### How to use this connector
```
Comment (JakeQuilty) [inline examples/generic_connector_configs/README.md:78]:
```suggestion
#### How to use this connector
```

File: examples/generic_connector_configs/README.md [hunk 9]
```diff
@@ -1,30 +1,232 @@
...
+* Edit the supplied configuration to get your [GitHub OAuth token](https://developer.github.com/v3/#oauth2-token-sent-in-a-header) from the correct provider/path.
+* Run Secretless with the supplied configuration
+* Query the GitHub API using `http_proxy=localhost:8081 curl api.github.com/{request}`
+
+### Example Usage
```
Comment (JakeQuilty) [inline examples/generic_connector_configs/README.md:83]:
```suggestion
#### Example Usage
```

File: examples/generic_connector_configs/README.md [hunk 10]
```diff
@@ -1,30 +1,232 @@
...
+    </li>
+    <code>
+      http_proxy=localhost:8081 curl -X GET api.github.com/users/username
+    </code>
+  </ol>
+</details>
+
+___
+
+## OAuth 2.0 API
```
Comment (JakeQuilty) [inline examples/generic_connector_configs/README.md:112]:
```suggestion
### OAuth 2.0 API
```

File: examples/generic_connector_configs/README.md [hunk 11]
```diff
@@ -1,30 +1,232 @@
...
+This generic OAuth HTTP connector can be used for any service that accepts a
+Bearer token as an authorization header.
+
+The configuration file for the OAuth 2.0 API can be found at
+[oauth2_secretless.yml](./oauth2_secretless.yml).
+
+### How to use this connector
```
Comment (JakeQuilty) [inline examples/generic_connector_configs/README.md:119]:
```suggestion
#### How to use this connector
```

File: examples/generic_connector_configs/README.md [hunk 12]
```diff
@@ -1,30 +1,232 @@
...
+* Edit the supplied service configuration to get your OAuth token
+* Run secretless with the supplied configuration(s)
+* Query the API using `http_proxy=localhost:8071 curl <Your OAuth2 API Endpoint URL>/{Request}`
+
+___
+
+## Slack Web API
```
Comment (JakeQuilty) [inline examples/generic_connector_configs/README.md:126]:
```suggestion
### Slack Web API
```

File: examples/generic_connector_configs/README.md [hunk 13]
```diff
@@ -1,30 +1,232 @@
...
+This example can be used to interact with [Slack's Web API](https://api.slack.com/apis).
+
+The configuration file for the Slack Web API can be found at [slack_secretless.yml](./slack_secretless.yml).
+
+### How to use this connector
```
Comment (JakeQuilty) [inline examples/generic_connector_configs/README.md:131]:
```suggestion
#### How to use this connector
```

File: examples/generic_connector_configs/README.md [hunk 14]
```diff
@@ -1,30 +1,232 @@
...
+* Edit the supplied configuration to get your Slack [OAuth token](https://api.slack.com/legacy/oauth#flow)
+* Run secretless with the supplied configuration(s)
+* Query the Slack API using `http_proxy=localhost:9030 curl -d {data} <Slack Endpoint URL>` or `http_proxy=localhost:9040 curl -d {data} <Slack Endpoint URL>`
+depending on if your endpoint requires JSON or URL encoded requests
+
+### Example Usage
```
Comment (JakeQuilty) [inline examples/generic_connector_configs/README.md:137]:
```suggestion
#### Example Usage
```

File: examples/generic_connector_configs/README.md [hunk 15]
```diff
@@ -1,30 +1,232 @@
...
+      -f examples/generic_connector_configs/slack_secretless.yml
+    </code>
+    <li>On another terminal window, make a request to Slack using Secretless</li>
+    <code>http_proxy=localhost:9030 curl -X POST --data '{"channel":"C061EG9SL","text":"I hope the tour went well"}' slack.com/api/chat.postMessage</code>
+  </ol>
+</details>
+
+___
+
+## Splunk API
```
Comment (JakeQuilty) [inline examples/generic_connector_configs/README.md:156]:
```suggestion
### Splunk API
```

File: examples/generic_connector_configs/README.md [hunk 16]
```diff
@@ -1,30 +1,232 @@
...
+This example can be used to interact with [Splunk's API](https://api.slack.com/apis).
+
+The configuration file for the Slack Web API can be found at [splunk_secretless.yml](./splunk_secretless.yml).
+
+### How to use this connector
```
Comment (JakeQuilty) [inline examples/generic_connector_configs/README.md:161]:
```suggestion
#### How to use this connector
```

File: examples/generic_connector_configs/README.md [hunk 17]
```diff
@@ -1,28 +1,221 @@
 # Generic HTTP Connector Example Configurations
 
+## Table of Contents
+- [Generic HTTP Connector Example Configurations](#generic-http-connector-example-configurations)
+  * [Table of Contents](#table-of-contents)
```
Comment (izgeri) [inline examples/generic_connector_configs/README.md:5]:
```suggestion
```
I think it's weird when the title of the page and the TOC are referenced in the TOC

File: examples/generic_connector_configs/README.md [hunk 18]
```diff
@@ -1,28 +1,221 @@
...
+- [Sample Configurations](#sample-configurations)
+  * [Elasticsearch API](#elasticsearch-api)
+  * [Github API](#github-api)
+  * [OAuth 2.0 API](#oauth-20-api)
+  * [Slack Web API](#slack-web-api)
+  * [Splunk API](#splunk-api)
+- [Contributing](#contributing)
+
+## Sample Configurations
```
Comment (izgeri) [inline examples/generic_connector_configs/README.md:14]: What comes below this reads more like an "Introduction". Can we make this section "Introduction" and then below it have a section for "Sample Configurations" that has a sentence or two to explain what's in the section and then jumps in to the sample configs?

**Pull request merged**